plugins/modules/consul_agent_check.py

```diff
@@ -0,0 +1,270 @@
+    notes: "Nginx Check"
+    
+
+- name: Remove check for service 'nginx'
+  community.general.consul_agent_check:
+    state: absent
+    id: nginx_http_check
+    service_id: "{{ nginx_service.ID }}"
+
+
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.consul import (
+    AUTH_ARGUMENTS_SPEC,
+    OPERATION_CREATE,
+    OPERATION_UPDATE,
+    OPERATION_DELETE,
+    OPERATION_READ,
+    _ConsulModule,
+    camel_case_key,
+    RequestError,
+    validate_check,
+)
+
+_ARGUMENT_SPEC = {
+    "state": dict(default="present", choices=["present", "absent"]),
+    "name": dict(type='str'),
+    "id": dict(type='str'),
+    "interval": dict(type='str'),
+    "notes": dict(type='str'),
+    "http": dict(type='str'),
+    "tcp": dict(type='str'),
+    "ttl": dict(type='str'),
+    "timeout": dict(type='str'),
+    "service_id": dict(type='str'),
+}
+
+_MUTUALLY_EXCLUSIVE = [
+    ('args', 'ttl', 'tcp', 'http'),
+]
+
+_REQUIRED_IF = [
+    ('state', 'present', ['name']),
+    ('state', 'absent', ['id']),
+]
+
+_REQUIRED_BY = {
+    'args': 'interval',
+    'http': 'interval',
+    'tcp': 'interval',
+}
+
+_ARGUMENT_SPEC.update(AUTH_ARGUMENTS_SPEC)
+
+
+class ConsulAgentCheckModule(_ConsulModule):
+    api_endpoint = "agent/check"
+    result_key = "check"
+    unique_identifier = "id"
+
+    def endpoint_url(self, operation, identifier=None):
+        if operation == OPERATION_READ:
+            return "/".join([self.api_endpoint + 's'])
+        if operation in [OPERATION_CREATE, OPERATION_UPDATE]:
+            return "/".join([self.api_endpoint, "register"])
+        if operation == OPERATION_DELETE:
+            return "/".join([self.api_endpoint, "deregister", identifier])
+
+        return super(ConsulAgentCheckModule, self).endpoint_url(operation, identifier)
+
+    def read_object(self):
+        url = self.endpoint_url(OPERATION_READ)
+        checks = self.get(url)
+
+        if self.params.get(self.unique_identifier) in checks:
+            return checks[self.params.get(self.unique_identifier)]
+        else:
+            return
+
+    def prepare_object(self, existing, obj):
+        existing = super(ConsulAgentCheckModule, self).prepare_object(existing, obj)
+
+        operational_attributes = {"Node", "CheckID", "Output", "ServiceName", "ServiceTags",
+                                  "Status", "Type", "ExposedPort", "Definition"}
+        existing = {
+            k: v for k, v in existing.items() if k not in operational_attributes
+        }
+
+        validate_check(existing)
+
+        if 'Args' in existing and existing['Args']:
+            existing['Args'] = ["sh", "-c", existing['Args']]
+
+        return existing
+
+    def create_object(self, obj):
+        super(ConsulAgentCheckModule, self).create_object(obj)
+        if self._module.check_mode:
+            return obj
+        else:
+            return self.read_object()
```

Comment:
The point of `create_object` is to create the object, why do you need to read the object here? `create_object` from the super class already returns a usable object, no?

Reply:
Because the Service/Check Register API Endpoint do not return a value I do the read again to return the created Service/Check

Reply:
ACK, the agent stuff is really weird. For instance when creating an auth method you get the new object returned: https://developer.hashicorp.com/consul/api-docs/acl/auth-methods#sample-response 

Guess the agent check/service predate that configuration.

Reply:
Yes I agree that it is realy weird. But this also happens in the catalog registry: https://developer.hashicorp.com/consul/api-docs/catalog#register-entity

Reply:
Would it be possible to move that into the `ConsulModule` base class and simply reread the object on create/update if the put returns `None`? Should allow you to get rid of all the `create_object`/`update_object` overrides.

Reply:
yes sounds good to me.

Reply:
Nice that looks clearly better. Can you think of any other improvements that we could add to `ConsulModule` so the handling of name vs id becomes easier in your case?

Reply:
I think I have a solution for it. I will push it in a minute